### PR TITLE
Backup unit tests

### PR DESCRIPTION
**Pull request by MiaAltieri**
Adding unit tests for `lib/charms/mongodb/v0/mongodb_backups.py`

### CODE REVIEW

File: tests/unit/test_mongodb_backups.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        resync.side_effect = ResyncError
+
+        self.harness.add_relation(RELATION_NAME, "s3-integrator")
+        self.harness.charm.backups._on_create_backup_action(action_event)
+
+        action_event.defer.assert_called()
+
+    @patch("charm.subprocess.check_output")
+    @patch("charm.MongoDBBackups._verify_resync")
+    def test_backup_failed(self, resync, subprocess):
```
Comment (marceloneppel) [inline tests/unit/test_mongodb_backups.py:115]: Is it possible to have a test that checks for a correct backup start? (by checking that `set_results` was called and/or `MaintenanceStatus` was set)

Comment (MiaAltieri) [reply tests/unit/test_mongodb_backups.py:115]: Those will be tested in the integration tests :)

**Comment by MiaAltieri** [comment]
Converting to DRAFT state -- need to add more unit tests after #153 merges

**Comment by MiaAltieri** [comment]
closing -- too many modifications made to the backup code. Tests need to be majorly updated

**Comment by MiaAltieri** [comment]
closing -- too many modifications made to the backup code. Tests need to be majorly updated

**Pull request closed without merging**